lib/barbeque/executor/hako.rb

```diff
@@ -41,6 +119,41 @@ def env_options(envs)
           "--env=#{key}=#{value}"
         end
       end
+
+      def s3_key_for_stopped_result(hako_task)
+        "#{@s3_prefix}/#{hako_task.task_arn}/stopped.json"
+      end
+
+      def s3_client
+        @s3_client ||= Aws::S3::Client.new(region: @s3_region, http_read_timeout: 5)
+      end
+
+      def get_stopped_result(hako_task)
+        object = s3_client.get_object(bucket: @s3_bucket, key: s3_key_for_stopped_result(hako_task))
+        JSON.parse(object.body.read)
```

Comment:
We need to know the whole string on JSON parse error.
When `extract_task_info` fails, we have to record the output of `hako oneshot --no-wait` command to get the whole string. The whole output of `hako oneshot --no-wait` will help us debug the exception.
When `get_stopped_result` fails, we can get the whole JSON-format string from S3.
That's why I added exception handing in `extract_task_info`.